spirv_msl.cpp

```diff
@@ -3246,15 +3408,16 @@ void CompilerMSL::emit_custom_functions()
 		spv_function_implementations.insert(SPVFuncImplForwardArgs);
 		spv_function_implementations.insert(SPVFuncImplGetSwizzle);
 	}
-
-	for (auto &spv_func : spv_function_implementations)
+	
+	for (const auto& spv_func : spv_function_implementations)
 	{
 		switch (spv_func)
 		{
 		case SPVFuncImplMod:
 			statement("// Implementation of the GLSL mod() function, which is slightly different than Metal fmod()");
 			statement("template<typename Tx, typename Ty>");
-			statement("inline Tx mod(Tx x, Ty y)");
+			statement(force_inline);
```

Comment:
Why does this have to be force inline?

Reply:
Seems to be unnecessary for templates. It was probably added to be on the safe side.

Reply:
I reverted static inlining for all template functions.